diff -r cf547c694dc5 setuptools/command/test.py
--- a/setuptools/command/test.py	Sat Jan 16 17:39:54 2016 -0500
+++ b/setuptools/command/test.py	Sat Jan 23 11:54:50 2016 +0100
@@ -151,12 +151,13 @@
         if self.distribution.tests_require:
             self.distribution.fetch_build_eggs(self.distribution.tests_require)
 
-        cmd = ' '.join(self._argv)
-        if self.dry_run:
-            self.announce('skipping "%s" (dry run)' % cmd)
-        else:
-            self.announce('running "%s"' % cmd)
-            self.with_project_on_sys_path(self.run_tests)
+        if self.test_suite:
+            cmd = ' '.join(self._argv)
+            if self.dry_run:
+                self.announce('skipping "%s" (dry run)' % cmd)
+            else:
+                self.announce('running "%s"' % cmd)
+                self.with_project_on_sys_path(self.run_tests)
 
     def run_tests(self):
         # Purge modules under test from sys.modules. The test loader will
